src/Cache/Engine/RedisEngine.php

```diff
@@ -228,14 +228,21 @@ public function clear($check)
         if ($check) {
             return true;
         }
-        $keys = $this->_Redis->getKeys($this->_config['prefix'] . '*');
 
-        $result = [];
-        foreach ($keys as $key) {
-            $result[] = $this->_Redis->del($key) > 0;
+        $this->_Redis->setOption(Redis::OPT_SCAN, Redis::SCAN_RETRY);
+
+        $isAllDeleted = true;
+        $iterator = null;
+        $pattern = $this->_config['prefix'] . '*';
+
+        while ($keys = $this->_Redis->scan($iterator, $pattern)) {
+            foreach ($keys as $key) {
```

Comment:
You could solve the warning from scrutinizer by doing

```php
while (true) {
  $keys = $this->_Redis->scan($iterator, $pattern)
  if ($keys === false) {
    break;
  }
  foreach ($keys as $key) {
  }
```

Reply:
Normally I'm not a fan of `while(true)` loops, but we should do everything we can to make Scrutinizer happy, right? ;-)